WooCommerce/Classes/ViewRelated/Orders/Order Creation/ProductsSection/AddProductToOrderViewModel.swift

```diff
@@ -29,31 +40,155 @@ final class AddProductToOrderViewModel: ObservableObject {
+    ///
+    private let syncingCoordinator = SyncingCoordinator()
+
+    /// Tracks if the infinite scroll indicator should be displayed
+    ///
+    @Published private(set) var shouldShowScrollIndicator = false
+
+    /// View models of the ghost rows used during the loading process.
+    ///
+    var ghostRows: [ProductRowViewModel] {
+        return Array(0..<6).map { index in
+            ProductRowViewModel(product: sampleGhostProduct(id: index), canChangeQuantity: false)
+        }
+    }
+
     /// Products Results Controller.
     ///
     private lazy var productsResultsController: ResultsController<StorageProduct> = {
-        let predicate = NSPredicate(format: "siteID == %lld AND statusKey IN %@ AND NOT(productTypeKey IN %@)",
-                                    siteID, includedProductStatusKeys, excludedProductTypeKeys)
+        let predicate = NSPredicate(format: "siteID == %lld", siteID)
         let descriptor = NSSortDescriptor(key: "name", ascending: true)
         let resultsController = ResultsController<StorageProduct>(storageManager: storageManager, matching: predicate, sortedBy: [descriptor])
         return resultsController
     }()
 
-    init(siteID: Int64, storageManager: StorageManagerType = ServiceLocator.storageManager) {
+    init(siteID: Int64, storageManager: StorageManagerType = ServiceLocator.storageManager, stores: StoresManager = ServiceLocator.stores) {
         self.siteID = siteID
         self.storageManager = storageManager
+        self.stores = stores
 
+        configureSyncingCoordinator()
         configureProductsResultsController()
     }
 }
 
+// MARK: - SyncingCoordinatorDelegate & Sync Methods
+extension AddProductToOrderViewModel: SyncingCoordinatorDelegate {
+    /// Sync products from remote.
+    ///
+    func sync(pageNumber: Int, pageSize: Int, reason: String? = nil, onCompletion: ((Bool) -> Void)?) {
+        transitionToSyncingState()
+        let action = ProductAction.synchronizeProducts(siteID: siteID,
+                                                       pageNumber: pageNumber,
+                                                       pageSize: pageSize,
+                                                       stockStatus: nil,
+                                                       productStatus: nil,
+                                                       productType: nil,
+                                                       productCategory: nil,
+                                                       sortOrder: .nameAscending,
+                                                       shouldDeleteStoredProductsOnFirstPage: false) { [weak self] result in
+            guard let self = self else { return }
+
+            switch result {
+            case .failure(let error):
+                DDLogError("⛔️ Error synchronizing products during order creation: \(error)")
+            case .success:
+                self.updateProductsResultsController()
```

Comment:
nit: `success` should come before `failure`

Reply:
Updated in 1e5d0d9.